pyigm/cgm/utils.py

```diff
@@ -84,6 +84,99 @@ def calc_rho(galaxy, igm_sys, cosmo, ang_sep=None, correct_lowz=True,
 
 
+def get_close_galaxies(field,rholim=300.*u.kpc,minz=0.001,maxz=None):
+    '''
+    Generate table of galaxies close to sightlines for one more IgmGalaxyField
+
+    Parameters
+    ----------
+    fields : IgmGalaxyField or list of IgmGalaxyField
+      Fields to be included
+    rholim : quantity, optional
+      Maximum impact parameter limit of galaxies
+    minz : float, optional
+      Minimum redshift of galaxies
+
+    Returns
+    -------
+    closegaltab : astropy Table
+      Table of galaxies meeting impact parameter and redshift criteria
+    '''
+    if maxz is None:
+        maxz = np.inf
+    field.galaxies['rho'] = field.calc_rhoimpact(field.galaxies,comoving=False)
+    idxs=np.where((field.galaxies['rho'].quantity<rholim)&
+                  (field.galaxies['Z']>minz)&(field.galaxies['Z']<maxz))[0]
+    closegaltab=field.galaxies[idxs]
+    if len(closegaltab) == 0:
+        print('No galaxies found meeting these selection criteria.')
+    return closegaltab
+
+
+def cgmabssys_from_sightline_field(field,sightline,rholim=300.*u.kpc,
+                                    minz=0.001,maxz=None,dv_max=400.*u.km/u.s):
+    """Instantiate list of CgmAbsSys objects from IgmgGalaxyField and IGMSightline.
+
+    Parameters
+    ----------
+    field : IgmGalaxyField
+    sightline : IGMSightline
+    rholim : Quantity, optional
+        Maximum impact parameter for associated galaxies
+    minz : float, optional
+        Minimum redshift for galaxy/absorber search
+    maxz : float, optional
+        Maximum redshift for galaxy/absorber search
+    dv_max : Quantity, optional
+        Maximum galaxy-absorber velocity separation
+
+    Returns
+    -------
+    cgmabslist : list
+        List of CgmAbsSys objects
+    """
+
+    closegals = get_close_galaxies(field,rholim,minz,maxz)
+    cgmabslist = []
+    for i,gal in enumerate(closegals):
+        galobj = Galaxy((gal['RA'],gal['DEC']),z=gal['Z'])
+        cgmobj = cgm_from_galaxy_igmsystems(galobj,sightline._abssystems,
+                                                dv_max=dv_max)
+        cgmabslist.extend(cgmobj)
+    return cgmabslist
+
+
+def cgmsurvey_from_sightlines_fields(fields,sightlines,name=None,**kwargs):
+    """Instantiate CGMAbsSurvey object from lists fo IgmGalaxyFields and IGMSightlines
+
+    Parameters
+    ----------
+    fields: list of IgmGalaxyFields
+    sightlines : list of IGMSightlines
+    name : str, optional
+        Name for the survey
+
+    Returns
+    -------
+    cgmsurvey: CGMAbsSurvey
+    """
+
+    if ((not isinstance(fields,list))|(not isinstance(sightlines,list))|
+        (len(fields) != len(sightlines))):
+        raise IOError("Inputs fields and sightlines must lists of the same length")
+
+    from pyigm.cgm.cgmsurvey import CGMAbsSurvey
+    cgmsys = []
+    for i,ff in enumerate(fields):
+        thiscgmlist = cgmabssys_from_sightline_field(ff,sightlines[i],**kwargs)
+        cgmsys.extend(thiscgmlist)
+    if name is not None:
```

Comment:
This seems a bit strange.  The way we instantiate
depends on whether the name variable is set?
Check this.

Reply:
Ah, the 'else' case should also instantiate as CGMAbsSurvey.from_cgmabssys() but without declaring a 'survey' argument.